torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -194,28 +194,28 @@ inline Tensor as_view(const Tensor & base, Tensor tensor, bool is_differentiable
     base_var = base_var._base();
   }
   if (is_differentiable) {
-    return make_variable_differentiable_view(std::move(base_var), std::move(tensor), creation_meta, std::move(view_func));
+    return make_variable_differentiable_view(std::move(base_var), tensor, creation_meta, std::move(view_func));
   } else {
     TORCH_CHECK(creation_meta == CreationMeta::DEFAULT,
                 "Non-differentiable views must have creation_meta=CreationMeta::DEFAULT");
-    return make_variable_non_differentiable_view(std::move(base_var), std::move(tensor));
+    return make_variable_non_differentiable_view(std::move(base_var), tensor);
   }
 }
 
 // See NOTE [ Autograd View Variables ] for details.
-inline std::vector<Tensor> as_view(const Tensor & base, std::vector<Tensor> tensors, bool is_differentiable,
+inline std::vector<Tensor> as_view(const Tensor & base, std::vector<Tensor>& tensors, bool is_differentiable,
```

Comment:
This looks really suspicious. Previously, a copy of `tensors` would have been made when you called `as_view`, so mutating it in the function is harmless. Now you are taking a mutable reference, which means you are silently mutating the caller's copy of tensors. Is this intended?

The old version of this function was implemented in a pretty suspicious way to (do a value-copy on entry, and then mutate this copy, and then return it), but if RVO worked (I don't know if it does in this case), the net number of copies is still 1, which is OK-ish.

Reply:
@ezyang  I feel mutating the caller's copy is intended. `tensors` are what we eventually return as results and we do want to set up view_meta on these tensors right? So I don't feel bad mutating these tensors since they're supposed to carry this information. 
In the previous implementation we made a copy of vector, but we actually didn't mutate it at all (other than move), but I think move is also redundant.

Reply:
@swolchok Not sure if this answers your question, the reason is `make_variable_differentiable_view` might either return caller's copy itself or return a new copy so that's why we do 
` tensor = make_variable_non_differentiable_view(base_var, tensor);` below.

Reply:
no, I meant the `tensors` argument to `as_view`. We both mutate it and copy & return it.

Reply:
oh I see what you mean now, yea we shouldn't copy... 
I'd just return a reference(instead of void) to keep codegen unchanged. fixing...